instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/LambdaImpl.java

```diff
@@ -0,0 +1,141 @@
+  }
+
+  private LambdaImpl() {}
+
+  @Nullable
+  static SdkRequest modifyRequest(
+      SdkRequest request, Context otelContext, TextMapPropagator messagingPropagator) {
+    System.out.println(
+        "LambdaImpl.modifyRequest " + request.getClass().getCanonicalName() + " from "
+            + request.getClass().getClassLoader() + "messagingPropagator" + messagingPropagator);
+
+    if (messagingPropagator == null) {
+      return null;
+    }
+
+    if (request instanceof InvokeRequest) {
+      System.out.println("injectIntoInvokeRequest");
+      return injectIntoInvokeRequest((InvokeRequest) request, otelContext, messagingPropagator);
+    } else {
+      return null;
+    }
+  }
+
+  private static SdkRequest injectIntoInvokeRequest(
+      InvokeRequest request, Context otelContext, TextMapPropagator messagingPropagator) {
+    String modifiedClientContext = injectIntoClientContext(request.clientContext(), otelContext,
+        messagingPropagator);
+    return request.toBuilder().clientContext(modifiedClientContext).build();
+  }
+
+  private static String injectIntoClientContext(
+      String originalClientContext,
+      Context otelContext,
+      TextMapPropagator messagingPropagator) {
+
+    try {
+      Map<String, Object> clientContext = deserialiseClientContext(originalClientContext);
+      if (clientContext == null) {
+        clientContext = new HashMap<>();
+      }
+
+      Object customObject = clientContext.computeIfAbsent("custom",
+          k -> new HashMap<String, String>());
+      if (!(customObject instanceof Map)) {
+        return originalClientContext;
+      }
+
+      @SuppressWarnings("unchecked")
+      Map<String, String> custom = (Map<String, String>) customObject;
+      messagingPropagator.inject(
+          otelContext,
+          custom,
+          MapSetter.INSTANCE
+      );
+      String modifiedClientContext = serialiseClientContext(clientContext);
+      // Make sure we don't exceed the size limit imposed by AWS https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html
+      if (modifiedClientContext.length() <= 3583) {
+        return modifiedClientContext;
+      } else {
+        return originalClientContext;
+      }
+    } catch (Throwable e) {
+      logger.warn("Failed to inject trace context into client context", e);
+      // Whenever something goes wrong we fall back to using the original clientContext
+      return originalClientContext;
+    }
+  }
+
+  @Nullable
+  private static Map<String, Object> deserialiseClientContext(String base64ClientContext)
+      throws Exception {
+    try {
+      if (base64ClientContext == null) {
+        return null;
+      } else {
+        String json = new String(Base64.getDecoder().decode(base64ClientContext), UTF_8);
+        TypeReference<HashMap<String, Object>> typeRef
+            = new TypeReference<HashMap<String, Object>>() {};
+        return OBJECT_MAPPER.readValue(json, typeRef);
```

Comment:
There is probably not much of a difference, but I think jackson can parse from bytes without an intermediate string.

Reply:
Thanks :+1: